### PR TITLE
Suite improvements for new lando (bug 1885346) (bug 1904696) (bug 1920987)

### PR DESCRIPTION
**Pull request by shtrom**

### CODE REVIEW

**Comment by shtrom** [comment]
@zzzeid this PR is a combination of
1. your new-lando PR #52 
2. suite improvements #59
3. (new) suite improvements for new lando

There may be a couple of conflicts when merging back into `main` due to changes to the lando- vs. lando-ui+lando-api -related, but it should just be a matter of allowing deletion of the old lando files.

File: docker-compose.yml
Comment (zzzeid) [inline docker-compose.yml:175]: As discussed, might be better to include the `setup_dev` command in the main lando container instead, along with the other development specific commands. Also, this could be combined in the main compose file.

**Comment by shtrom** [comment]
Ok, I moved things around, and did some cleanup while I was at it. I added a healthcheck, and needed to add a dependency on phabricator.test before starting lando, as it tries (and fails) to talk to it on start and gets into a restart loop.

Starting with the override,

```docker-compose -f docker-compose.yml -f docker-compose.bmo.yml -f docker-compose.lando.yml up -d```

works reliably (but is a bit slow due to the phabricator dep).

I tested without the override, by using the `suite_lando` image (after hacking it so it had the version file and static assets already installed) in the base compose file, and that booted fine, too, so we should be OK (:

**Review by zzzeid — APPROVED** [review]
LGTM. Will merge it now, and if we need to fine tune any of the changes we can do so before we merge #52 in.

**Pull request merged**
